my-react-app/src/components/GetData/GetData.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from "react";
+import DataItem from "./DataItem";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+type Data = {
+  id: number;
+  title: string;
+  year: string;
+  genre: string;
+  director: string;
+  actors: string;
+  language: string;
+  production: string;
+  website: string;
+};
+
+const GetData = () => {
+  const postQuery = useQuery({
+    queryKey: ["posts"],
+    queryFn: async () => {
+      const response = await axios.get("https://freetestapi.com/api/v1/movies");
```

Comment:
sehr gut gemacht jetzt verwenden wir react query noch um die daten invalide zu schalten sobald der user ein modal zu macht 

react query speichert die geladenen daten ja in einem art cache zwischen, damit die app nicht immer die daten neu laden muss, wenn man weis dass sich an den daten sowieso nichts geändert hat. jetzt wollen wir aber dem package mitteilen dass sich eventuell an den daten was verändert hat. das können wir indem wir der query sagen dass die daten invalide sind, react query holt sich die aktualisierten dann von ganz alleine.. 

einfach mal googlen und die docu durschschauen.. sollte nicht so schwer sein umzusetzen aber ganz wichtiges feature von tanstack